UnicornRender/video/include/unicorn/video/Camera.hpp

```diff
@@ -8,141 +8,172 @@
-    UNICORN_EXPORT const glm::mat4& GetView() const;
-
-    /**
-     * @brief Returns field of view value
-     * @return Field of view value
-     */
+    /** @brief Sets near Z plane */
+    UNICORN_EXPORT void SetZNear(float znear);
+
+    /** @brief Sets far Z plane */
+    UNICORN_EXPORT void SetZFar(float zfar);
+
+    /** @brief Returns fov */
     UNICORN_EXPORT float GetFov() const;
+private:
+    float m_fov, m_fovLowerBound, m_fovUpperBound, m_znear, m_zfar;
 
-    /**
-    * @brief Checks if view or projection need to be updated and update it, if needed
-    */
-    UNICORN_EXPORT void Frame();
+    void UpdateProjection() override final;
+};
+
+/** @brief Orthographic camera projection controller */
+class OrthographicCamera : public CameraProjection
+{
+public:
+    UNICORN_EXPORT OrthographicCamera(system::Window* window, glm::mat4& cameraProj);
+    
+    UNICORN_EXPORT void UpdateProjection() override final;
+    UNICORN_EXPORT void Scroll(float yoffset) override;
+    UNICORN_EXPORT void SetScale(float scale);
+
+    UNICORN_EXPORT float GetScale() const;
 private:
-    float m_aspect;
-    glm::vec3 m_position;
-    glm::vec3 m_upVector;
-    glm::vec3 m_direction;
-
-    struct
-    {
-        glm::mat4 m_projection;
-        glm::mat4 m_view;
-    } m_matrices;
-
-    float m_fov;
-    float m_znear, m_zfar;
-    bool m_dirtyView;
-    float m_fovLowerBound;
-    float m_fovUpperBound;
-    bool m_dirtyProjection;
-    /**
-     * @brief Calculates view matrix
-     */
-    void UpdateViewMatrix();
-    /**
-     * @brief Calculates projection matrix
-     */
-    void UpdateProjectionMatrix();
+    float m_orthoScale;
+    float m_scaledAspect;
+    float m_right, m_top;
+    float m_back, m_front;
 };
+
+/** @brief 2D camera controller */
+class Camera2DController : public CameraController
+{
+public:
+    UNICORN_EXPORT Camera2DController(glm::mat4& cameraView);
+
+    UNICORN_EXPORT void MoveUp(float deltaTime);
+    UNICORN_EXPORT void MoveDown(float deltaTime);
+    UNICORN_EXPORT void MoveLeft(float deltaTime);
+    UNICORN_EXPORT void MoveRight(float deltaTime);
+private:
+};
+
+/** @brief FPS style camera controller */
+class CameraFpsController : public CameraController
+{
+public:
+    UNICORN_EXPORT CameraFpsController(glm::mat4& cameraView);
+    UNICORN_EXPORT void MoveUp(float deltaTime);
+    UNICORN_EXPORT void MoveDown(float deltaTime);
+    UNICORN_EXPORT void MoveLeft(float deltaTime);
+    UNICORN_EXPORT void MoveRight(float deltaTime);
+    UNICORN_EXPORT void MoveForward(float deltaTime);
+    UNICORN_EXPORT void MoveBackward(float deltaTime);
+
+    /** @brief Updates view matrix by taking diff between past and new mouse coordinates */
+    UNICORN_EXPORT void UpdateView(double posX, double posY);
+
+    /** @brief Sets mouse coordinates without updating view matrix */
+    UNICORN_EXPORT void UpdateViewPositions(double posX, double posY);
+
+    //! Sensitivity of mouse
+    float sensitivity;
```

Comment:
Move member declaration to the top of the list, please.

Reply:
Nope